### ISSUE TITLE
Random issue with AddSlider

### ISSUE THREAD

**Issue by AndyBeamish** [piotrulos/MSCModLoader#57]
#### Info
* **OS version/build:** Windows 10 27244.165
* **Architecture:** 64 bit
* **MSCLoader version:**  v0.43
* **Type of issue:** bug/crash/suggestion (remove unnecessary)

#### Description
Mods using Slider settings sometimes, when exiting the game back to the spinning car screen, appear to attempt to load with the wrong type. (Including the font slider for the console.) Issue does not affect all sliders at once just one or more seemingly randomly.
#### Attachments
Please attach **output_log.txt** from `mysummercar_Data` folder  
Screenshoots related to your issue.
![20180723043556_1](https://user-images.githubusercontent.com/25791244/43056228-d31ba138-8e32-11e8-9804-562d7a6a1c60.jpg)
[output_log.txt](https://github.com/piotrulos/MSCModLoader/files/2217861/output_log.txt)

**Comment by piotrulos** [comment]
Yes, you need run debug.bat each time you compile your mod. But you can use "post-build script" to make this script automatic. (as written [here](https://github.com/piotrulos/MSCModLoader/wiki/Debugging-your-mods-in-Visual-Studio))

I actually never hit that bug, but it doesn't mean that it didn't exists. And since this is random it will be harder to catch. Maybe some other mod messing with this? Try disable other mods and see if this will still happen.

**Comment by AndyBeamish** [comment]
I'll have a play about with the code and hopefully get back to you with a fix.

**Comment by piotrulos** [comment]
I'll check also if there is fix needed to mod loader, maybe something is not unloading correctly when going back to menu. 
Meanwhile you should try to use int cast before value, maybe it will force code to use int.
for example: `(int)100`

**Comment by piotrulos** [comment]
I see how to fix that on mod loader level, I need to add Single as potential int value.
It actually never happens to float because foats have "f" at the end of value.
But int was detected as int or double, looks like you hit thrid posibility that int value can be detected as Single.

**Comment by piotrulos** [comment]
Try this MSCLoader.dll file [MSCLoader.zip](https://github.com/piotrulos/MSCModLoader/files/2220488/MSCLoader.zip) and test if this will fix that problem

**Comment by AndyBeamish** [comment]
Yay! I think I'll need to do some casts on the values in my mod to get everything working 100% but the MSCLoader side is working great, many thanks.

**Comment by AndyBeamish** [comment]
BTW... can I package the experimental MSCLoader with my mod?

**Comment by piotrulos** [comment]
No need, I will package normal release soon. I just want to release also an updater so I'm doing some final testing.
